File: Python/100_SameTree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

def DFS (p: TreeNode, q:TreeNode):
    if p.val != q.val:
        return False
    
    if p.left == None and p.right == None and q.right == None and q.left == None:
        return True
    
    b = True
    
    ## Check left
    if p.left != None and q.left != None:
         b = DFS(p.left, q.left)
    elif not (p.left == None and q.left == None):
        return False
    
    if b == False:
        return b
    
    ## Check right
    if p.right != None and q.right != None:
        b = DFS(p.right, q.right)
    elif not (p.right == None and q.right == None):
        return False
            
    return b

class Solution:
    def isSameTree(self, p: TreeNode, q: TreeNode) -> bool:
        if p == None and q != None:
            return False
        if p != None and q == None:
            return False
        if p == None and q == None:
            return True
        
        return DFS(p,q)
        
        

File: C#/100_SameTree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public bool IsSameTree(TreeNode p, TreeNode q) {
        List<int> pTraverse = new List<int>();
        Queue<TreeNode> queue = new Queue<TreeNode>();
        queue.Enqueue(p);
        while (queue.Count > 0) {
            TreeNode v = queue.Dequeue();
            pTraverse.Add(v == null? -1 : v.val);
            if (v == null) continue;
            queue.Enqueue(v.left);
            queue.Enqueue(v.right);
        }
        // Checking next tree..
        queue.Enqueue(q);
        int c = 0;
        while (queue.Count > 0) {
            TreeNode v = queue.Dequeue();
            int vVal = v == null? -1 : v.val;
            if (pTraverse[c] != vVal) return false;
            c++;
            if (v == null) continue;
            queue.Enqueue(v.left);
            queue.Enqueue(v.right);
        }
        return true;
    }
}
